lib/promote_release.js

```diff
@@ -147,23 +147,31 @@ export default class ReleasePromotion extends Session {
       cli.warn(`Aborting release promotion for version ${version}`);
       throw new Error('Aborted');
     }
-    await this.cherryPickToDefaultBranch();
+    const appliedCleanly = await this.cherryPickToDefaultBranch();
```

Comment:
I think it's better to check if the release is `semver-patch` instead of `appliedCleanly`. It will be confusing to people when reading the code.

Reply:
I don't think it's a good idea to assume that a semver-patch release never conflicts, and that a non-semver-patch would always conflict, that would be setting ourselves to fail. I don't understand why it would be confusing

Reply:
I feel confusing the `git checkout HEAD^` | `git checkout HEAD` part considering a `appliedCleanly` variable. Maybe a comment? I'm fine leaving it as `appliedCleanly` with a comment on that part

Reply:
We could use `git rev-parse HEAD` before trying to cherry-pick, and then `git checkout` (or the less ambiguous `git restore`) with the exact commit sha.

Reply:
I decided to add comments